prow/tide/tide.go

```diff
@@ -179,18 +183,100 @@ func byRepoAndNumber(prs []PullRequest) map[string]PullRequest {
 	return m
 }
 
+// requirementDiff calculates the diff between a PR and a TideQuery.
+// This diff is defined with a string that describes some subset of the
+// differences and an integer counting the total number of differences.
+// Note: an empty diff can be returned if the reason that the PR does not match
+// the TideQuery is unknown. This can happen happen if this function's logic
+// does not match GitHub's and does not indicate that the PR matches the query.
+func requirementDiff(pr *PullRequest, q *config.TideQuery) (string, int) {
```

Comment:
TODO: we should coordinate this logic with the PR status page and possible do this computation in prow instead of in the frontend.